contracts/AcceleratingDistributor.sol

```diff
@@ -147,17 +147,21 @@ contract AcceleratingDistributor is ReentrancyGuard, Ownable, Multicall {
     }
 
     /**
-     * @notice Recover an ERC20 token either dropped on the contract or excess after the staking program ends.
-     * @dev Any wallet can call this function as it will only ever send tokens to the owner of the distributor.
-     * @param tokenAddress The address of the token to recover.
-     * @param amount The amount of the token to recover.
+     * @notice Enables the owner to recover tokens dropped onto the contract. This could be used to remove unclaimed
+     * staking rewards or recover excess LP tokens that were inadvertently dropped onto the contract. Importantly, the
+     * contract will only let the owner recover staked excess tokens above what the contract thinks it should have. i.e
+     * the owner cant use this method to steal staked tokens, only recover excess ones mistakenly sent to the contract.
+     * @param token The address of the token to skim.
      */
-    function recoverErc20(address tokenAddress, uint256 amount) external {
-        require(stakingTokens[tokenAddress].lastUpdateTime == 0, "Can't recover staking token");
-        require(tokenAddress != address(rewardToken), "Can't recover reward token");
-        IERC20(tokenAddress).safeTransfer(owner(), amount);
-
-        emit RecoverErc20(tokenAddress, owner(), amount, msg.sender);
+    function recoverToken(address token, uint256 amount) external onlyOwner {
+        // If the token is an enabled staking token then we want to preform a skim action where we send back any extra
+        // tokens that are not accounted for in the cumulativeStaked variable. This lets the owner recover extra tokens
+        // sent to the contract that were not explicitly staked. if the token is not enabled for staking then we simply
+        // send back the full amount of tokens specified by the caller.
+        if (stakingTokens[token].lastUpdateTime != 0) amount -= stakingTokens[token].cumulativeStaked;
```

Comment:
As per OZ's comments: 

"The owner would be able to steal staking tokens if they pass "stakingTokens[token].cumulativeStaked * 2" as the "amount" argument. Have you all considered not taking in an amount value, but instead just skimming off the balance over the staked amount?"

thoughts?